### DISCUSSION TITLE
Support redis cluster

### DISCUSSION THREAD

**Discussion by sbanditster** [directus/directus#17743] [Feature Requests]
### Summary

Please add the possibility to connect to a redis cluster. At the moment directus can only connect to a non clustered redis cluster and it gives an error, when I try to connect to clustered redis.
The disadvantage of this solution, that the non clustered redis have only 1 master and directus can only connect to the master, because it reads and write data. However if the redis master node restarted e.g.: because of kured in kubernetes cluster or any other reason, the directus content is not available.

### Basic Example

error while trying to connect to a redis cluster:

`ReplyError: MOVED 14387 10.2.0.157:6379 at parseError (/directus/node_modules/redis-parser/lib/parser.js:179:12) at parseType (/directus/node_modules/redis-parser/lib/parser.js:302:14) { command: { name: 'get', args: [ 'system-cache_lock:system-cache-lock' ] } }`

### Motivation

When using multi replica directus setup. e.g.: 3 or more directus pods in a kubernetes cluster for more performance and high availability, external cache must be used.
If I use directus internal cache in this case, it leads to data inconsistency.

### Detailed Design

It is also discussed here:
https://github.com/directus/directus/pull/6131

### Must Have:
- Ability to connect to a redis cluster with directus.

### Should Have:
- e.g.: new env variables for redis cluster connection.

### Drawbacks

n/a

### Alternatives

What other designs have been considered?
If directus could connect to the slave nodes in a non clustered redis cluster not just the master, it would be also a solution.

What is the impact of not doing this?
Cannot use HA caching solution.

### Adoption Strategy

It should be a new feature?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by sgreinerCNS** [comment]
We are also interested into this. We want to have multiple directus nodes across two datacenters. Also we want to (or better to say have to) use a redis cluster in each datacenter to eliminate a Single Point of Failure. I understand, that smaller setups do not essentially need this, but as a larger customer this might be the blocking reason to not go with or buy directus.

**Reply by rijkvanzanten** [reply]
Working on it! 🙂 This PR https://github.com/directus/directus/pull/20514 replaces our current keyv based setup for caching etc which in turn allows us to start relying on more powerful redis specific features, including adding support for proper clustering / read-write split etc
